packages/cli/src/commands/site/deploy/index.ts

```diff
@@ -14,54 +14,126 @@
  * limitations under the License.
  */
 
-import type { CommandModule } from "yargs";
+import { consola } from "consola";
+import type * as yargs from "yargs";
+import type { FoundryConfig, SiteConfig } from "../../../util/config.js";
+import configLoader from "../../../util/configLoader.js";
 import type { CommonSiteArgs } from "../CommonSiteArgs.js";
 import type { SiteDeployArgs } from "./SiteDeployArgs.js";
 
-export const command: CommandModule<
-  CommonSiteArgs,
-  SiteDeployArgs
-> = {
-  command: "deploy",
-  describe: "Deploy an uploaded version",
-  builder: (argv) => {
-    return argv
-      .options({
-        "siteVersion": {
-          type: "string",
-          conflicts: "clearVersion",
-          // group: "Deploy Version",
-          // implies: { "clearVersion": "false" },
-        },
-        undeploy: {
-          alias: "clearVersion",
-          description: "Causes the site to no longer be deployed",
-          type: "boolean",
-          conflicts: "siteVersion",
-          // group: "Deploy Version",
-          // implies: { "siteVersion": "" },
-        },
-      }).group(
-        ["siteVersion", "clearVersion"],
-        "Version To Deploy (requires one of)",
-      );
+async function deployHandler(): Promise<
```

Comment:
Is it possible to keep these as constants like they were before and move the async config loading into the builder? For example `.builder()` accepts returning a `PromiseLike`